lib/jobs/ipmi-job.js

```diff
@@ -312,42 +228,161 @@ function ipmiJobFactory(
-                    return results[0];
-                }else{
+            waterline.workitems.findOne({id: data.workItemId})
+        ]).spread(function (selInfo, workObj) {
+                workObject = workObj;
+                totalEntries = parseInt(selInfo.Entries);
+                /*  Parses SEL info output:
+                 "SEL Information
+                 Version          : 1.5 (v1.5, v2 compliant)
+                 Entries          : 187
+                 Free Space       : 13008 bytes
+                 Percent Used     : 18%
+                 Last Add Time    : 01/01/1970 01:56:22
+                 Last Del Time    : Not Available
+                 Overflow         : false
+                 Supported Cmds   : 'Delete' 'Reserve'"
+                 */
+                lastPreviouslyReadSelEntry = _.get(workObj, 'config.lastPreviouslyReadSelEntry', "0000");// jshint ignore:line
+
+                var tmpDate = selInfo["Last Del Time"].split(/\/|:| /);
+                var tmpDate2 = tmpDate[0] + " " + tmpDate[1] + ", " + tmpDate[2] +
+                    " " + tmpDate[3] + ":" + tmpDate[4] + ":" + tmpDate[5];
+                lastSelDeleteTimeCurrent = new Date(tmpDate2);
+                lastSelDeleteTimeLastRun = _.get(workObj, 'config.lastSelDeleteTimeLastRun');
+
+                lastSelDeleteTimeLastRun = new Date(lastSelDeleteTimeLastRun);
+
+                if (selInfo["Last Del Time"] === "Not Available" ||
+                    (lastSelDeleteTimeLastRun === undefined && lastSelDeleteTimeCurrent !== undefined) ||
+                    lastSelDeleteTimeCurrent.getTime() <= lastSelDeleteTimeLastRun.getTime()) {
+                    
+                    selCleared = false;
+
+                } else {
+                    selCleared = true;
+                    //Sel log was cleared and  we reset the first record to be be read at 0
+                    logger.info('****SEL was cleared. lastSelDeleteTimeCurrent is :' +
+                        lastSelDeleteTimeCurrent + "and lastSelDeleteTimeLastRun is " +
+                        lastSelDeleteTimeLastRun +". Poller: "+data.workItemId+ " nodeID: " + workObject.node);
+
+                    workObject.config.lastPreviouslyReadSelEntry = "0000";
+                    workObject.config.lastSelDeleteTimeLastRun = lastSelDeleteTimeCurrent;
+                    return workObject.save();
+                }
+            })
+            .then(function(){
+                if (totalEntries !== 0 && selCleared === false ) {
+                    //It skips one poll when the SEL is being cleared
+                    var readFile = Promise.promisify(fs.readFile);
+                    var unlink = Promise.promisify(fs.unlink);
+                    temp = Promise.promisifyAll(temp);
+                    var filePath;
+                    return temp.openAsync('tmp-ipmi-data')
+                        .then(function (info) {
+                            return info.path;
```

Comment:
There needs to be a line inserted before the "return info.path":
fs.close(info.fd);

(the temp module doesn't actually track the file itself and clean up when it goes out of scope, so if we don't do this, we will leak all the file descriptors).